### ISSUE TITLE
https.get doesn't use https.request

### ISSUE THREAD

**Issue by RomanGotsiy** [substack/https-browserify#6]
`https.get` doesn't use overloaded `https.request`. Instead it uses `http.request`.
This is caused by the fact that `http.get` uses directly `http.request`: https://github.com/substack/http-browserify/blob/master/index.js#L44 and don't know anything about overloaded version.
Need to overload `https.get` method as well.

**Comment by RomanGotsiy** [comment]
Just notices that there is already PR (#1) that fixes this.
@substack is there a reason not to merge this PR?

**Comment by feross** [comment]
Are you using the latest version of `browserify` v13?

Browserify doesn't use `http-browserify` for `require('http')` anymore. See [`stream-http`](https://github.com/jhiesey/stream-http) instead.

**Comment by 0815fox** [comment]
This info should be mentioned in ReadMe.md - I just fell over the same thing.

So neither http-browserify nor https-browserify should be used anymore?

**Comment by feross** [comment]
You should just use `require('http')` and let browserify take care of the rest.

**Comment by RomanGotsiy** [comment]
> Are you using the latest version of browserify v13?

@feross this doesn't change the fact this lib is buggy.
Closing this issue makes illusion the lib works fine.
Other people may copy this code, or other people may stick on the same issue.

I strongly believe that this bug should be opened just to allow other people see it.

**Comment by 0815fox** [comment]
Totally agree with @RomanGotsiy - at the very least it should be mentioned, that this lib should not be used anymore. At best, this Ticket remains open, so that others can see that it is not fixed in the code in this repo.

**Comment by RomanGotsiy** [comment]
@feross please reopen this issue...

This issue is not solved. Other people should see it. I constantly face issues with this other libs use it..

Also, it would be perfect to add DEPRECATED or NOT MAINTAINED warning on REAMDE

**Comment by terinjokes** [comment]
@RomanGotsiy What bug actually happens here? You never explain a bug. In a bundle `http` as used by `https-browserify` is `stream-http`, and the browser is what actually setups the connection.

**Comment by feross** [comment]
Yeah, I'm really confused. Please open a new issue with example code that demonstrates an actual bug.

This library is not deprecated. It's still used by browserify.

**Comment by RomanGotsiy** [comment]
I can add an example to this bug. The title of this bug tells exactly what the bug is about.

When `https.get` method is called it doesn't use overloaded `https.request` ([this code](https://github.com/substack/https-browserify/blob/master/index.js#L10-L13) is not used at all).
Instead `http.request` method is called.

The simplest example:

```js
const https = require("https");
const options = {
  host: 'api.github.com',
  method: 'GET',
  withCredentials: false
};
https.get('https://api.github.com');
```

Run this code in the browser and check the browser network tab. The actual request will be done using `HTTP` not `HTTPS`:
![image](https://cloud.githubusercontent.com/assets/3975738/18231472/134fd0f4-72c3-11e6-9987-8f6b5af130f0.png)


Plunker: http://plnkr.co/edit/2pRjbLNWtp0fSVLYMKPc?p=preview

**Comment by feross** [comment]
@RomanGotsiy Thanks for clarifying.

@jhiesey Do you have any ideas for how to fix this elegantly? Should we just enumerate all the http functions so we don't need to depend on `stream-http` internals?

**Comment by terinjokes** [comment]
@feross: that would be my preference.

**Comment by RomanGotsiy** [comment]
@feross, @jhiesey, do you have any estimates when this will be fixed?
There already was a PR solving the issue but it was closed: https://github.com/substack/https-browserify/pull/1/files

Maybe I can help somehow?

I still have issues because of this lib. Both Webpack and browserify use `stream-http` for `http` BUT for `https` they use `https-browserify` ([browserify](https://github.com/jhiesey/stream-http/issues/24#issuecomment-140529233), [webpack](https://github.com/webpack/node-libs-browser/blob/master/package.json#L19)) which is broken.

cc @TheLarkInn

**Comment by feross** [comment]
@RomanGotsiy Please send a PR that fixes the issue and I'll take a look at it.

**Comment by jhiesey** [comment]
@RomanGotsiy If you don't get to it first I'll look into it this weekend.

**Comment by RomanGotsiy** [comment]
@feross PR #8

**Comment by jhiesey** [comment]
Alright, it looks like there's a PR to fix this with some comments on it, so I'll leave this to @feross and @RomanGotsiy unless my help is needed.

**Comment by feross** [comment]
Thanks for the PR, @RomanGotsiy. I fixed it up a bit so it behaves exactly the same as Node.js (to my knowledge). Commit here:

Published as 1.0.0.

**Comment by feross** [comment]
Also, released browserify 14.3.0. https://github.com/substack/node-browserify/blob/master/changelog.markdown#1430

**Comment by RomanGotsiy** [comment]
Awesome! Thank you!
